components/suggestion/search_user_provider.tsx

```diff
@@ -75,20 +81,21 @@ class SearchUserSuggestion extends Suggestion {
 }
 
 export default class SearchUserProvider extends Provider {
-    constructor(userSearchFunc) {
+    autocompleteUsersInTeam: (username: string) => any;
+    constructor(userSearchFunc: (username: string) => DispatchFunc) {
```

Comment:
If you track back `userSearchFunc` func, you will see that it is returning `UserAutocomplete` as seen below.

https://github.com/mattermost/mattermost-webapp/blob/7d527c9c852f9e931cf253c16f663371340738d5/packages/client/src/client4.ts#L913-L916

Therefore, this change should do the trick. 
```suggestion
    autocompleteUsersInTeam: (username: string) => Promise<UserAutocomplete>;
    constructor(userSearchFunc: (username: string) => Promise<UserAutocomplete>) {
```

You also need to change the return type of `autocompleteUsersInTeam` in line below 
https://github.com/mattermost/mattermost-webapp/blob/7d527c9c852f9e931cf253c16f663371340738d5/components/search/types.ts#L54